packages/@uppy/provider-views/src/View.ts

```diff
@@ -15,39 +95,42 @@ export default class View {
     this.cancelPicking = this.cancelPicking.bind(this)
   }
 
-  preFirstRender () {
+  preFirstRender(): void {
     this.plugin.setPluginState({ didFirstRender: true })
     this.plugin.onFirstRender()
   }
 
-  // eslint-disable-next-line class-methods-use-this
-  shouldHandleScroll (event) {
-    const { scrollHeight, scrollTop, offsetHeight } = event.target
+  shouldHandleScroll(event: Event): boolean {
+    const { scrollHeight, scrollTop, offsetHeight } =
+      event.target as HTMLElement
     const scrollPosition = scrollHeight - (scrollTop + offsetHeight)
 
     return scrollPosition < 50 && !this.isHandlingScroll
   }
 
-  clearSelection () {
+  clearSelection(): void {
     this.plugin.setPluginState({ currentSelection: [], filterInput: '' })
   }
 
-  cancelPicking () {
+  cancelPicking(): void {
     this.clearSelection()
 
     const dashboard = this.plugin.uppy.getPlugin('Dashboard')
 
     if (dashboard) {
+      // @ts-expect-error impossible to type this correctly without adding dashboard
+      // as a dependency to this package.
       dashboard.hideAllPanels()
     }
   }
 
-  handleError (error) {
+  handleError(error: Error): void {
     const { uppy } = this.plugin
     const message = uppy.i18n('companionError')
 
     uppy.log(error.toString())
 
+    // @ts-expect-error AuthError lives in @uppy/companion-client
     if (error.isAuthError || error.cause?.name === 'AbortError') {
```

Comment:
`@ts-expect-error` hides ALL the errors, using assertions you at least retain some type checking
```suggestion
    if ((error as any).isAuthError || (error.cause as Error)?.name === 'AbortError') {
```